src/pages/Script/ScriptsComp.tsx

```diff
@@ -58,6 +67,12 @@ export const ScriptTransactions = ({ page, size }: { page: number; size: number
     history.push(`/${language}/script/${codeHash}/${hashType}?page=${page}&size=${size}`)
   }
 
+  const isChecked = restrict === true
+  const switchRestrictMode = (checked: boolean) => {
+    setRestrict(checked)
+    history.push(`/${language}/script/${codeHash}/${hashType}`)
```

Comment:
The `restrict` is set in the query key, and the data will be refetched automatically, so `history.push` is unnecessary.